### PR TITLE
Fix threshold_li initial guess in 0.14

### LINKED ISSUE #4140: threshold_li fails on array with 2 values in it
## Description
For an array with only 2 values in it, threshold_li fails; this worked in 0.14.2, but fails in 0.14.3 and 0.14.4 .  This was causing 3 tests in CellProfiler to fail; I've bypassed it for two tests that were not in fact supposed to use threshold_li (which was our bad), but the third is still failing.  It's unlikely such images would exist directly off a microscope, but if someone fed in a binary mask (ala example "a" below), it would fail, which is definitely an issue.

## Way to reproduce
```python
a = numpy.array([0, 0, 1, 0, 0, 1, 0, 1])
b = numpy.array([0, 0, 0.1, 0, 0, 0.1, 0, 0.1])
c = numpy.array([0, 0, 0.1, 0, 0, 0.1, 0.01, 0.1])
d = numpy.array([0, 0, 1, 0, 0, 1, 0.5, 1])
e = numpy.array([1,1])
f = numpy.array([1,2])
skimage.filters.threshold_li(a)
skimage.filters.threshold_li(b)
skimage.filters.threshold_li(c)
skimage.filters.threshold_li(d)
skimage.filters.threshold_li(e)
skimage.filters.threshold_li(f)
```
This comes out this way in 0.14.2:

```
>>> skimage.filters.threshold_li(a)
/Users/bcimini/.virtualenvs/skimageli/lib/python2.7/site-packages/skimage/filters/thresholding.py:601: RuntimeWarning: divide by zero encountered in log
  (np.log(mean_back) - np.log(mean_fore)))
0.0
>>> skimage.filters.threshold_li(b)
0.0
>>> skimage.filters.threshold_li(c)
0.02505097742626249
>>> skimage.filters.threshold_li(d)
0.0
>>> skimage.filters.threshold_li(e)
1
>>> skimage.filters.threshold_li(f)
1.0
```

But this way in 0.14.3/4:

```
>>> skimage.filters.threshold_li(a)
/Users/bcimini/.virtualenvs/skimageli/lib/python2.7/site-packages/numpy/core/fromnumeric.py:3118: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/Users/bcimini/.virtualenvs/skimageli/lib/python2.7/site-packages/numpy/core/_methods.py:85: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
nan
>>> skimage.filters.threshold_li(b)
nan
>>> skimage.filters.threshold_li(c)
0.02505097742626249
>>> skimage.filters.threshold_li(d)
/Users/bcimini/.virtualenvs/skimageli/lib/python2.7/site-packages/skimage/filters/thresholding.py:596: RuntimeWarning: divide by zero encountered in log
  (np.log(mean_back) - np.log(mean_fore)))
0.0
>>> skimage.filters.threshold_li(e)
1
>>> skimage.filters.threshold_li(f)
nan
```

## Version information
```python
2.7.12 (default, Aug 27 2019, 14:36:02) 
[GCC 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.10.44.4)]
Darwin-18.7.0-x86_64-i386-64bit
scikit-image version: 0.14.4
numpy version: 1.16.5
```

### PR DESCRIPTION
**Pull request by jni**
## Description

Fixes #4140 

## Checklist

- [ ] ~~[Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)~~
- [ ] ~~Gallery example in `./doc/examples` (new features only)~~
- [ ] ~~Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark~~
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/thresholding.py [hunk 1]
```diff
@@ -581,8 +581,8 @@ def threshold_li(image):
     tolerance = np.min(np.diff(np.unique(image))) / 2
 
     # Initial estimate
-    t_curr = np.mean(image)
```
Comment (hmaarrfk) [inline skimage/filters/thresholding.py]: Can we add a comment here

Comment (jni) [reply skimage/filters/thresholding.py]: @hmaarrfk ask and ye shall receive. =) Thanks for the quick review!

**Comment by jni** [comment]
Guuuuuh... @hmaarrfk any ideas on how to fix the build errors? I see four:

- array pad `test_mode_not_set`. I think this was fixed on master but I can't find the PR...

```
=================================== FAILURES ===================================
________________________ ValueError3.test_mode_not_set _________________________
self = <skimage.util.tests.test_arraypad.ValueError3 testMethod=test_mode_not_set>

    def test_mode_not_set(self):
        arr = np.arange(30).reshape(5, 6)
        with testing.raises(TypeError):
>           pad(arr, 4)

E           Failed: DID NOT RAISE <class 'TypeError'>

skimage/util/tests/test_arraypad.py:996: Failed
```
- QtCore.PYQT_VERSION_STR doesn't exist (this is in mpl):

```
_________________ ERROR collecting io/tests/test_mpl_imshow.py _________________
skimage/io/tests/test_mpl_imshow.py:6: in <module>

    import matplotlib.pyplot as plt
../../../venv/lib/python2.7/site-packages/matplotlib/pyplot.py:115: in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
../../../venv/lib/python2.7/site-packages/matplotlib/backends/__init__.py:63: in pylab_setup
    [backend_name], 0)
../../../venv/lib/python2.7/site-packages/matplotlib/backends/backend_qt4agg.py:9: in <module>
    from .backend_qt5agg import (
../../../venv/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py:15: in <module>
    from .backend_qt5 import (
../../../venv/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py:19: in <module>
    import matplotlib.backends.qt_editor.figureoptions as figureoptions
../../../venv/lib/python2.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py:20: in <module>
    import matplotlib.backends.qt_editor.formlayout as formlayout
../../../venv/lib/python2.7/site-packages/matplotlib/backends/qt_editor/formlayout.py:54: in <module>
    from matplotlib.backends.qt_compat import QtGui, QtWidgets, QtCore
../../../venv/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py:147: in <module>
    _setup_pyqt4()
../../../venv/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py:126: in _setup_pyqt4
    _setup_pyqt4_internal(api=2)
../../../venv/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py:115: in _setup_pyqt4_internal
    __version__ = QtCore.PYQT_VERSION_STR

E   AttributeError: 'module' object has no attribute 'PYQT_VERSION_STR'
```

- RAG is gone for some reason???

```
_______________ ERROR collecting future/graph/tests/test_rag.py ________________
ImportError while importing test module '/home/travis/build/scikit-image/scikit-image/skimage/future/graph/tests/test_rag.py'.

Hint: make sure your test modules/packages have valid Python names.

Traceback:
skimage/future/graph/tests/test_rag.py:2: in <module>
    from skimage.future import graph
skimage/future/__init__.py:8: in <module>
    from . import graph
E   ImportError: cannot import name graph
```

- pytest crashes during test_plugins

I've seen the TypeError and the plugins one before, but don't remember where and can't find the corresponding PR. The other two are new and I don't have any ideas...

**Comment by hmaarrfk** [comment]
can you update the travis default. I think travis changed the default parameters and updated the version of ubuntu it was using. Should help.

**Comment by jni** [comment]
@hmaarrfk what do you mean by updating the travis default? i went through the travis file and decided to remove 3.4 and 3.5, but didn't find anything else to update (except maybe removing the gitter stuff but I don't think that's the problem...)

File: .travis.yml [hunk 2]
```diff
@@ -5,6 +5,9 @@
 
 # See tools/travis/notes.txt for some guidelines
 
+# If this is ever changed to xenial (Ubuntu 16.04), make sure to enable the
+# xvfb service
+dist: trusty
```
Comment (hmaarrfk) [inline .travis.yml]: @jni

**Comment by hmaarrfk** [comment]
I think you also have to bump minimum requirements for python 3.6

**Comment by hmaarrfk** [comment]
I guess you need

https://github.com/scikit-image/scikit-image/commit/6e560620fd1913e4953dff0e4a1cd4bf4ae04b37

**Comment by hmaarrfk** [comment]
Gotta delete 3.5 from appvetor matrix. Travis failed out of luck

**Comment by hmaarrfk** [comment]
ok, this is good to go. it also updates the 14.4 build requirements.

**Comment by jni** [comment]
@hmaarrfk thank you!!! :pray: :pray: :pray: 

@scikit-image/core if someone else can review + merge this after CI completes, I can push out 0.14.5!

File: skimage/util/tests/test_arraypad.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from __future__ import division, absolute_import, print_function
+from distutils.version import LooseVersion
 
 import numpy as np
+# note: skimage.util.pad is just numpy.pad
```
Comment (sciunto) [inline skimage/util/tests/test_arraypad.py]: then, why not calling np.pad directly?

Comment (jni) [reply skimage/util/tests/test_arraypad.py]: (a) this is a change from master backported to make Travis happy, so best to ignore it.
(b) we have historically wrapped pad to harmonize behaviour across numpy versions. Not sure whether it's still needed but not about to fix this in this PR

Comment (sciunto) [reply skimage/util/tests/test_arraypad.py]: Thanks for the explanation.

**Pull request merged**